src/components/VideoLibraryModal.vue

```diff
@@ -0,0 +1,594 @@
+            <button
+              v-for="button in menuButtons"
+              :key="button.name"
+              :disabled="!button.enable"
+              class="flex flex-col justify-center align-center"
+              @click="currentTab = button.name.toLowerCase()"
+            >
+              <v-tooltip v-if="button.tooltip !== ''" open-delay="600" activator="parent" location="top">{{
+                button.tooltip
+              }}</v-tooltip>
+              <div
+                class="rounded-full mb-1 text-2xl"
+                :class="[
+                  button.enable ? 'frostedButton' : 'frostedButtonDisabled',
+                  currentTab === button.name.toLowerCase() ? 'w-[60px] h-[60px]' : 'w-[40px] h-[40px]',
+                ]"
+              >
+                <v-icon
+                  :size="currentTab === button.name.toLowerCase() ? 40 : 24"
+                  :class="{ 'ml-1': button.name.toLowerCase() === 'videos' }"
+                  >{{ button.icon }}</v-icon
+                >
+              </div>
+              <div class="text-sm" :class="{ 'text-white/30': !button.enable }">
+                {{ button.name }}
+              </div>
+            </button>
+          </div>
+          <div>
+            <v-divider class="opacity-[0.08]"></v-divider>
+            <button class="flex flex-col justify-center py-2 mt-4 align-center" @click="closeModal">
+              <div class="frostedButton flex flex-col justify-center align-center w-[28px] h-[28px] rounded-full mb-1">
+                <v-icon class="text-[18px]">mdi-close</v-icon>
+              </div>
+              <div class="text-sm">Close</div>
+            </button>
+          </div>
+        </div>
+        <v-divider vertical class="h-[92%] mt-4 opacity-[0.08]"></v-divider>
+        <!--------------------------->
+        <!-- TAB for Video Button  -->
+        <template v-if="currentTab === 'videos'">
+          <!-- Available Videos -->
+          <div
+            v-if="availableVideosAndLogs.length > 0"
+            class="flex flex-col justify-between align-center py-8 px-2 w-[300px] h-[500px]"
+          >
+            <div class="flex flex-col align-center h-full w-full overflow-auto px-4">
+              <div v-for="video in availableVideosAndLogs" :key="video.filename">
+                <video
+                  v-if="video.filename.endsWith('.webm')"
+                  :id="`video-library-${video.filename}`"
+                  class="border-4 border-white rounded-md cursor-pointer border-opacity-[0.1] hover:border-opacity-[0.4] transition duration-75 hover:ease-in"
+                  :class="[
+                    selectedVideos.find((v) => v.filename === video.filename)
+                      ? 'border-opacity-[0.4]'
+                      : 'border-opacity-[0.1]',
+                    selectedVideos.find((v) => v.filename === video.filename) ? 'w-[220px]' : 'w-[190px]',
+                  ]"
```

Comment:
Couldn't we use something like ?
```suggestion
                  :class="selectedVideos.find((v) => v.filename === video.filename)
                    ? ['border-opacity-[0.4]', 'w-[220px]']
                    : ['border-opacity-[0.1]', 'w-[190px]']
                  "
```